### DISCUSSION TITLE
How does one "mask" in 7.x? (6.x examples use deprecated -mask option)

### DISCUSSION THREAD

**Discussion by CelestialStreamer** [ImageMagick/ImageMagick#6756] [Help]
### ImageMagick version

7.1.0-25

### Operating system, version and so on

Windows 10

### Description

I am only becoming more aware of how the internet is chock full of 6.x examples of imagemagick code, most of which works with 7.x. One feature I thought would be useful is `-mask`. Any time I use it:
```
magick.exe: option deprecated, unable to execute `-mask' at CLI arg 15 @ error/operation.c/CLIOption/5418
```

Looking through the [documentation](https://imagemagick.org/script/command-line-options.php), `-clip` and `-clip-mask` look promising, but I haven't found a modern example online yet.

Here are some use cases I'd like to use "masks" for:

* Define an area programatically with stuff like `-polygon ....` or whatever.
* From an image `mask.png` (please explain white vs black and/or transparent pixels?)
* From an image built in code. Think, threshold an image and use black/white areas as mask.

**Comment by snibgo** [comment]
Yes, the documentation eg https://imagemagick.org/Usage/masking/#masks needs updating for v7.

Where v6 has "-mask" and "+mask", use v7 "-write-mask" and "+write-mask".

I use "-write-mask" fairly often. I don't think I've ever used "-read-mask".
